packages/server/lib/StudioLifecycleManager.ts

```diff
@@ -0,0 +1,156 @@
+    debugData: any
+    ctx: DataContext
+  }): void {
+    debug('Initializing studio manager')
+
+    const studioManagerPromise = getAndInitializeStudioManager({
+      projectId,
+      cloudDataSource,
+    }).then(async (studioManager) => {
+      if (studioManager.status === 'ENABLED') {
+        debug('Cloud studio is enabled - setting up protocol')
+        const protocolManager = new ProtocolManager()
+        const protocolUrl = routes.apiRoutes.captureProtocolCurrent()
+        const script = await api.getCaptureProtocolScript(protocolUrl)
+
+        await protocolManager.prepareProtocol(script, {
+          runId: 'studio',
+          projectId: cfg.projectId,
+          testingType: cfg.testingType,
+          cloudApi: {
+            url: routes.apiUrl,
+            retryWithBackoff: api.retryWithBackoff,
+            requestPromise: api.rp,
+          },
+          projectConfig: _.pick(cfg, ['devServerPublicPathRoute', 'port', 'proxyUrl', 'namespace']),
+          mountVersion: api.runnerCapabilities.protocolMountVersion,
+          debugData,
+          mode: 'studio',
+        })
+
+        studioManager.protocolManager = protocolManager
+      } else {
+        debug('Cloud studio is not enabled - skipping protocol setup')
+      }
+
+      debug('Studio is ready')
+      this.studioReady = true
+      this.studioManager = studioManager
+      this.callRegisteredListeners()
+
+      return studioManager
+    }).catch(async (error) => {
+      debug('Error during studio manager setup: %o', error)
+
+      const cloudEnv = (process.env.CYPRESS_INTERNAL_ENV || 'production') as 'development' | 'staging' | 'production'
+      const cloudUrl = ctx.cloud.getCloudUrl(cloudEnv)
+      const cloudHeaders = await ctx.cloud.additionalHeaders()
+
+      reportStudioError({
+        cloudApi: {
+          cloudUrl,
+          cloudHeaders,
+          CloudRequest,
+          isRetryableError,
+          asyncRetry,
+        },
+        studioHash: projectId,
+        projectSlug: cfg.projectId,
+        error,
+        studioMethod: 'initializeStudioManager',
+        studioMethodArgs: [],
+      })
+
+      // Clean up any registered listeners
+      this.listeners = []
+
+      return null
+    })
+
+    this.studioManagerPromise = studioManagerPromise
+
+    // Register this instance in the data context
+    ctx.update((data) => {
+      data.studioLifecycleManager = this
+    })
+  }
+
+  isStudioReady (): boolean {
+    return this.studioReady
+  }
+
+  async getStudio () {
+    if (!this.studioManagerPromise) {
+      throw new Error('Studio manager has not been initialized')
+    }
+
+    return await this.studioManagerPromise
+  }
+
+  private callRegisteredListeners () {
+    if (!this.studioManager) {
+      throw new Error('Studio manager has not been initialized')
+    }
+
+    debug('Calling all studio ready listeners')
+    this.listeners.forEach((listener) => {
+      listener(this.studioManager as StudioManager)
```

Comment:
Why does this need to be cast?

Reply:
Added a suggestion that I think should resolve the typescript issue you were seeing.